var/spack/repos/builtin/packages/cuda/package.py

```diff
@@ -465,6 +465,13 @@ def setup_build_environment(self, env):
     def setup_dependent_build_environment(self, env, dependent_spec):
         env.set("CUDAHOSTCXX", dependent_spec.package.compiler.cxx)
 
+    @property
+    def cmake_search_paths(self):
+        if self.spec.satisfies("target=x86_64:"):
+            cub_path = self.prefix.targets + "/x86_64-linux/lib/cmake"
+            return [cub_path]
+        return []
```

Comment:
If you make the changes suggested above, then this would need to be
```suggestion
        cmake_prefix_paths = [self.prefix]
        if self.spec.satisfies("target=x86_64:"):
            cub_path = self.prefix.targets + "/x86_64-linux/lib/cmake"
            cmake_prefix_paths.append(cub_path)
        return cmake_prefix_paths
```

Reply:
I was actually worried about taking this approach because I don't want to require that the user remembers to add `self.prefix`. I assume that (a) it never hurts to add this and (b) if it's generally needed that the user would be annoyed having to remember it every time.

If that doesn't seem like a problem to you, I can make this change.

Reply:
I think it's easier for the user to understand "list the cmake prefix paths needed to findd the package" than "list the cmake prefix paths needed to run this package, except the one we've decided is obvious".

Reply:
@becker33 we already have a few magic defaults like this.  For example configure_args or cmake_args adds the `--prefix` or `-DCMAKE_INSTALL_PREFIX_PATH` option by default.  I think it better for this function to return a list of additional cmake_prefix_paths.

Reply:
To be fair, as far as I know for most packages, the default does the right thing.  I imagine there are others (also probably closed source packages developed by Nvidia), but we wouldn't be adding this functionality now (as opposed to years ago) if more packages need this to work properly so I don't want to bikeshed getting this fixed for the small number of packages that actually need this.